File: medium/strings/Simplify Path.py
class Solution(object):
    def simplifyPath(self, path):
        r = list()
        for i in path.split("/"):
            if r!=[] and i=="..":
                r.pop()
            elif i not in [".","..",""]:
                r.append(i)
        return "/"+"/".join(r)
            
        
            
            

        

File: medium/strings/Simplify Path.java
class Solution {
    public String simplifyPath(String path) {
        var r = new ArrayList<String>();
        for(var i : path.split("/")){
            if(i.isEmpty() || i.equals(".")) continue;
            if(i.equals("..")){
                if(!r.isEmpty()){
                    r.remove(r.size()-1);
                }
            }
            else r.add(i);
        }return "/" + String.join("/",r);
    }
}